app/policies/article_policy.rb

```diff
@@ -144,13 +144,44 @@ def subscriptions?
     user_author? || user_super_admin?
   end
 
+  def elevated_user?
+    user_any_admin? || user_moderator?
+  end
+
   # this method performs the same checks that determine
   # if the record can be featured
   def revoke_publication?
     require_user!
     return false unless @record.published?
 
-    user_any_admin? || user_moderator?
+    elevated_user?
+  end
+
+  def allow_tag_adjustment?
+    require_user!
+
+    elevated_user? || should_show_adjust_tags(@record)
+  end
+
+  # If this comes back false, the tags that display
+  # are just those that the user is a tag-mod for
+  def adjust_all_tags?
+    require_user!
+
+    elevated_user?
+  end
+
+  def should_show_adjust_tags(record)
+    @tag_moderator_tags = Tag.with_role(:tag_moderator, @user)
```

Comment:
It doesn't look like we're using these instance variables in the class. Could we convert them to local variables (e.g. `@tag_moderator_tags` becomes `tag_moderator_tags`).

One potential gotcha is if there is any views that are expecting `@tag_moderator_tags` or `@adjustments`.

And yowza this logic.